### PR TITLE
Kak gate fix

### PR DESCRIPTION
**Pull request by rrealmuto**
Fixes a recently introduced issue where the Kak Gate setting doesn't work when set to "Open" if you don't start with zelda's letter. Also makes it so that if you find the letter while in the Kak scene with the setting set to "Zelda's Letter Opens Gate", the gate will open automatically and not require reloading the scene.

Should probably be tested with all of the different combinations of this setting + starting w/ letter + the dungeon reward shuffle which is what broke this to begin with.

https://github.com/OoTRandomizer/OoT-Randomizer/assets/5086838/ae1ed386-9d97-47a8-ad10-5ae5f8c787cc

### CODE REVIEW

**Comment by rrealmuto** [comment]
There's still a little weird behavior where if you find the letter in kak and then talk to the guard, he still thinks you haven't handed it in. You can still hand it in and then the cutscene plays. The gate will remain open.

File: ASM/c/item_effects.c [hunk 1]
```diff
@@ -223,12 +224,27 @@ void clear_excess_hearts(z64_file_t* save, int16_t arg1, int16_t arg2) {
...
         save->inf_table[7] = save->inf_table[7] | 0x40; // "Spoke to Gate Guard About Mask Shop"
         if (!COMPLETE_MASK_QUEST) {
             save->item_get_inf[2] = save->item_get_inf[2] & 0xFB87; // Unset "Obtained Mask" flags just in case of savewarp before Impa.
         }
+        // Check if we're in kak and actually open the gate
+        if (z64_game.scene_index == 82) {
+            // Loop through the actors looking for the gate
+            z64_actor_t* curr = z64_game.actor_list[7].first;
+            while(curr != NULL) {
+                if(curr->actor_id == 0x100) { // Check for BG_GATE_SHUTTER
```
Comment (fenhl) [inline ASM/c/item_effects.c]:
```suggestion
            while (curr != NULL) {
                if (curr->actor_id == 0x100) { // Check for BG_GATE_SHUTTER
```

File: ASM/src/bg_gate_shutter.asm [hunk 2]
```diff
@@ -0,0 +1,9 @@
+; Check the new gate open flag instead of the master sword check
+; This happens after the zelda's letter check so it should only effect it when the setting is set to open
+; Put the flag into t9 because it will be checked against 0 when we return
+; we can use t8
+bg_gate_shutter_open_hack:
+    li  t8, OPEN_KAKARIKO
+    lb  t8, 0x00(t8) ; read the value of the OPEN_KAKARIKO setting. If we set it to 2 then it's always open
+    jr  ra
```
Comment (fenhl) [inline ASM/src/bg_gate_shutter.asm]:
```suggestion
    li      t8, OPEN_KAKARIKO
    lb      t8, 0x00(t8) ; read the value of the OPEN_KAKARIKO setting. If we set it to 2 then it's always open
    jr      ra
```

**Review by fenhl — CHANGES_REQUESTED** [review]
Just some code style nits, seems good other than that.

**Review by fenhl — APPROVED** [review]

**Pull request merged**